core/SymbolRef.h

```diff
@@ -70,420 +107,420 @@ class Symbols {
 
 public:
     static SymbolRef noSymbol() {
-        return SymbolRef(nullptr, 0);
+        return SymbolRef();
     }
 
     static SymbolRef top() {
-        return SymbolRef(nullptr, 1);
+        return SymbolRef(nullptr, SymbolRef::Kind::ClassOrModule, 1);
```

Comment:
can we confirm that this compiles down to a statically known u4?

Reply:
How would I do that? Also, this constructor is defined in `Symbols.cc` -- would that optimization happen in linker optimizations?

Reply:
Looks like yes, at least in this one file implementation: https://godbolt.org/z/qdHvrK

I don't know how the heuristics change when the implementation of the function is defined in a separate compilation unit (that `SymbolRef` constructor is defined in `Symbols.cc`).